core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -0,0 +1,301 @@
+			tc.telemClient = telemPb.NewTelemClient(conn)
+		}
+
+		// Wait for close
+		<-tc.chDone
+
+	}()
+}
+
+// getCSAPrivateKey gets the client's CSA private key
+func (tc *telemetryIngressBatchClient) getCSAPrivateKey() (privkey []byte, err error) {
+	keys, err := tc.ks.GetAll()
+	if err != nil {
+		return privkey, err
+	}
+	if len(keys) < 1 {
+		return privkey, errors.New("CSA key does not exist")
+	}
+
+	return keys[0].Raw(), nil
+}
+
+// Send directs incoming telmetry messages to the worker responsible for pushing it to
+// the ingress server. If the worker telemetry buffer is full, messages are dropped
+// and a warning is logged.
+func (tc *telemetryIngressBatchClient) Send(payload TelemPayload) {
+	worker := tc.findOrCreateWorker(payload)
+	select {
+	case worker.chTelemetry <- payload:
+		worker.dropMessageCount.Store(0)
+	case <-payload.Ctx.Done():
+		return
+	default:
+		worker.logBufferFullWithExpBackoff(payload)
+	}
+}
+
+// findOrCreateWorker finds a worker by ContractID or creates a new one if none exists
+func (tc *telemetryIngressBatchClient) findOrCreateWorker(payload TelemPayload) *telemetryIngressBatchWorker {
+	tc.workersMutex.Lock()
+	defer tc.workersMutex.Unlock()
+
+	worker, found := tc.workers[payload.ContractID]
+
+	if !found {
+		worker = NewTelemetryIngressBatchWorker(
+			tc.telemMaxBatchSize,
+			tc.telemSendInterval,
+			tc.telemClient,
+			&tc.wgDone,
+			tc.chDone,
+			make(chan TelemPayload, tc.telemBufferSize),
+			payload.ContractID,
+			tc.globalLogger,
+			tc.logging,
+		)
+		worker.Start()
+		tc.workers[payload.ContractID] = worker
+	}
+
+	return worker
+}
+
+// telemetryIngressBatchWorker pushes telemetry in batches to the ingress server via wsrpc.
+// A worker is created per ContractID.
+type telemetryIngressBatchWorker struct {
+	telemMaxBatchSize uint
+	telemSendInterval time.Duration
+	telemClient       telemPb.TelemClient
+	wgDone            *sync.WaitGroup
+	chDone            chan struct{}
+	chTelemetry       chan TelemPayload
+	contractID        string
+	logging           bool
+	lggr              logger.Logger
+	dropMessageCount  atomic.Uint32
+}
+
+// NewTelemetryIngressBatchWorker returns a worker for a given contractID that can send
+// telemetry to the ingress server via WSRPC
+func NewTelemetryIngressBatchWorker(
+	telemMaxBatchSize uint,
+	telemSendInterval time.Duration,
+	telemClient telemPb.TelemClient,
+	wgDone *sync.WaitGroup,
+	chDone chan struct{},
+	chTelemetry chan TelemPayload,
+	contractID string,
+	globalLogger logger.Logger,
+	logging bool,
+) *telemetryIngressBatchWorker {
+	return &telemetryIngressBatchWorker{
+		telemSendInterval: telemSendInterval,
+		telemMaxBatchSize: telemMaxBatchSize,
+		telemClient:       telemClient,
+		wgDone:            wgDone,
+		chDone:            chDone,
+		chTelemetry:       chTelemetry,
+		contractID:        contractID,
+		logging:           logging,
+		lggr:              globalLogger.Named("TelemetryIngressBatchWorker"),
```

Comment:
We should have all names for all named loggers in one place here: https://github.com/smartcontractkit/chainlink/blob/567932f0bc793a5ca1804a6dfa40863793748769/core/logger/logger.go#L100

Reply:
Added the named logger as a constant